docs/reference/aggregations/metrics/top-metrics-aggregation.asciidoc

```diff
@@ -0,0 +1,206 @@
+[role="xpack"]
+[testenv="basic"]
+[[search-aggregations-metrics-top-metrics]]
+=== Top Metrics Aggregation
+
+experimental[We expect to change the response format of this aggregation as we add more features.]
+
+The `top_metrics` aggregation selects metrics from the document with the largest or smallest "sort"
+value. This gets the value of the `v` field on the document with the largest value of `s`:
+
+[source,console,id=search-aggregations-metrics-top-metrics-simple]
+----
+POST /test/_bulk?refresh
+{"index": {}}
+{"s": 1, "v": 3.1415}
+{"index": {}}
+{"s": 2, "v": 1}
+{"index": {}}
+{"s": 3, "v": 2.71828}
+POST /test/_search?filter_path=aggregations
+{
+  "aggs": {
+    "tm": {
+      "top_metrics": {
+        "metric": {"field": "v"},
```

Comment:
Do we expect most use-cases to select a single metric of multiple ones? If a single metric is the main use case, we could just take a field or a script and this would give a more consistent experience with other aggregations? If we expect the main use-case to want to retrieve multiple metrics, ignore this comment.

Reply:
I expect folks will want multiple metrics, yeah. But I'm not *sure* about it. Thus the experimental tag on this agg.

Reply:
I also wonder whether users will want to sort on this aggregation. I haven't looked in depth, but at the moment I think this aggregation doesn't allow its parent agg to be sorted by it?

Reply:
> I also wonder whether users will want to sort on this aggregation. I haven't looked in depth, but at the moment I think this aggregation doesn't allow its parent agg to be sorted by it?

It doesn't. I looked at it and it seems possible, but a bit fiddly because of some calls that the aggregation framwork makes that don't necessarily make sense for this agg. I propose that we wait on implementing this until later, if at all.

Reply:
I got it!